client/state/terms/selectors.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import get from 'lodash/get';
+import values from 'lodash/values';
+
+/**
+ * Returns terms for a site, filtered by taxonomy.
+ *
+ * @param  {Object} state    Global state tree
+ * @param  {Number} siteId   Site ID
+ * @param  {String} taxonomy Taxonomy slug
+ * @return {Array}           Terms
+ */
+export function getSiteTaxonomyTerms( state, siteId, taxonomy ) {
+	const terms = get( state.terms, [ siteId, taxonomy ], null );
```

Comment:
The fallback value here isn't being used, so I'd suggest either eliminate it or change the condition below to:

```js
if ( ! terms ) {
	return terms;
}
```